README.md

```diff
@@ -53,6 +53,8 @@ The key concept is the `MachineLearningProvider`. To implement an OpenML provide
 
 * To train new machine learning algorithms within the Feedzai platform, implement the `TrainingMachineLearningProvider` extension of the `MachineLearningProvider` interface.
 
+* Finally, generate a Jar with the code (possibly with all the dependencies necessary in it, or a set of Jars instead), and make sure you include META-INF to indicate which providers you have in your code. Check out [our example in this repository](https://github.com/feedzai/feedzai-openml/blob/master/openml-example/src/main/resources/META-INF/services/com.feedzai.mlapi.provider.MachineLearningProvider).
```

Comment:
I think we are lacking the explanation as why would someone need to add the META-INF on the jar file. While it's an implementation detail, we can't hide that fact from OpenML developers, and as such I feel we own them the explanation that the Feedzai platform (pulse) will look for any provider through a java ServiceLoader, hence the need for this declaration file

Reply:
I already say that this is to "make sure ... indicate which providers you have". That should be more than enough without getting into dirty details, no?

Reply:
true, I can live with hiding those details (that we are using a ServiceLoader and so on). But I truly believe that we should be more specific on this point: There should be a file under `resources/META-INF/services/com.feedzai.openml.MachineLearningProvider` (not sure if it has to be under resources?) with the fully qualified name of all providers, one per line

Reply:
Honestly ServiceLoader is such a well known standard/mechanism that I think it's easier to just to link to that and say that the service interface is `com.feedzai.openml.MachineLearningProvider` than to have to describe that it is a file in META-INF with name x and contents y...

Reply:
Just a side note here: Google's [Auto Service](https://github.com/google/auto/tree/master/service) uses annotation processing to generate the META-INF resources. IMO it's a good tip for people who are not used to dealing with java's Service Loader.